packages/grid/_modules_/grid/components/columnHeaders/GridColumnHeaderItem.tsx

```diff
@@ -197,7 +198,7 @@ export function GridColumnHeaderItem(props: GridColumnHeaderItemProps) {
       data-field={column.field}
       style={{
         width,
-        minWidth: width,
+        minWidth: minWidth !== undefined ? minWidth : width,
```

Comment:
This min-width on the element has nothing to do with the min-width of #1850. See the benchmark link to better capture what the problem is.